src/libs/kata-types/src/container.rs

```diff
@@ -136,6 +138,41 @@ impl State {
     }
 }
 
+/// Update annotations: Remove specified keys or add new key-value pairs.
+pub fn update_ocispec_annotations(
+    annotations: &HashMap<String, String, RandomState>,
+    keys_to_remove: &[&str],
+    keys_to_add: &[(String, String)],
+    allow_overwrite: bool,
```

Comment:
I'm a bit skeptical of boolean flags (e.g. https://dev.to/56_kode/clean-code-why-boolean-flags-in-function-parameters-are-a-code-smell-3fjp), and it looks like it's only being called with `allow_overwrite: false`. Should we just remove it?